### PR TITLE
Allow user to send a bug report

### PR DESCRIPTION
**Pull request by kntsoriano**

### CODE REVIEW

File: app/javascript/components/modals/BugReportModal.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    setStatus(BugReportModalStatus.INITIALIZED)
+  }, [open])
+
+  return (
+    <Modal isOpen={open} onRequestClose={onClose}>
+      <Status status={status} />
+      <form onSubmit={handleSubmit}>
+        <label htmlFor="body">Report</label>
+        <textarea id="body"></textarea>
+        <button type="submit">Submit</button>
```
Comment (SleeplessByte) [inline app/javascript/components/modals/BugReportModal.tsx]: Maybe disable this if `url` isn't present?

File: db/migrate/20201214073537_create_bug_reports.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateBugReports < ActiveRecord::Migration[6.1]
+  def change
+    create_table :bug_reports do |t|
+      t.belongs_to :user, null: false
+      t.text :body, null: false
```
Comment (iHiD) [inline db/migrate/20201214073537_create_bug_reports.rb]: Rename to `content` pls. Also, let's make this markdown, so can you rename to `content_markdown` and add a field for `content_html` which gets set in the `before_update` block pls?

File: db/migrate/20201214073537_create_bug_reports.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class CreateBugReports < ActiveRecord::Migration[6.1]
+  def change
+    create_table :bug_reports do |t|
+      t.belongs_to :user, null: false
```
Comment (iHiD) [inline db/migrate/20201214073537_create_bug_reports.rb]:
```suggestion
      t.belongs_to :user, null: false, foreign_key: true
```

**Review by iHiD — APPROVED** [review]

File: app/javascript/components/modals/BugReportModal.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+          if (!isMountedRef.current) {
+            throw new Error('Component not mounted')
+          }
+
+          setStatus(BugReportModalStatus.SUCCEEDED)
+
+          return json
+        })
+        .catch((err) => {
+          if (err.message === 'Component not mounted') {
```
Comment (ErikSchierboom) [inline app/javascript/components/modals/BugReportModal.tsx]: I know this is unlikely to change, but checking an error message and conditionally executing code based on that message feels slightly brittle to me. Will this only match when the error is thrown in the `then` clause above?

**Review by ErikSchierboom — APPROVED** [review]

File: app/javascript/components/modals/BugReportModal.tsx (hunk 4 above)
Comment (kntsoriano) [reply app/javascript/components/modals/BugReportModal.tsx]: I agree. I will refactor this one.

**Comment by iHiD** [comment]
👋

File: app/javascript/components/modals/BugReportModal.tsx (hunk 4 above)
Comment (SleeplessByte) [reply app/javascript/components/modals/BugReportModal.tsx]: Use a custom error, and then you can `instanceof`

**Pull request merged**